src/createButton.ts

```diff
@@ -12,11 +14,26 @@ export interface ButtonFactory extends ComposeFactory<Button, ButtonOptions> { }
 
 const createButton: ButtonFactory = createWidget
 	.mixin({
-		mixin: createFormFieldMixin
+		mixin: createFormFieldMixin,
+		aspectAdvice: {
+			before: {
+				getNodeAttributes(this: Button, overrides: VNodeProperties = {}) {
+					if (this.state.classes) {
+						const classes: { [index: string]: boolean; } = {};
+						this.state.classes.forEach((c) => classes[c] = true);
+						classes[ css.button ] = true;
+						overrides.classes = classes;
+					}
+
+					return [overrides];
+				}
+			}
+		}
 	})
 	.extend({
 		tagName: 'button',
-		type: 'button'
+		type: 'button',
+		classes: [ css.button ]
```

Comment:
@agubler and @matt-gadd talked about this.  We think that actually using the feature in compose (see: dojo/compose#46) that for classes that are "bound" to the factory/class, that this is the right way to add these and they will should be merged and added to every render.  What we need to do though is make `classes` `readonly` so that you cannot modify instance `classes` in this fashion and only look for instance bound classes in `.state` when rendering.

So for now this is the right way, though we need to make other changes to preserve the functionality...